hamilton/include/hamilton/hybrid_planner.h

```diff
@@ -0,0 +1,195 @@
+  std::string world_frame;                /* The name of the world link in the URDF file */
+  std::vector<std::string> joint_names;   /* A list with the names of the mobile joints in the robot */
+
+  /* Trajectory Generation Members:
+   *  Used to control the attributes (points, shape, size, etc) of the robot trajectory. */
+
+  //TODO add support for straight lines, circle, B-splines. add to list. 
+
+  double time_delay;              /* Time step between consecutive points in the robot path */
+  std::vector<double> seed_pose;  /* Joint values close to the desired start of the robot path */
+
+  /* Visualization Members
+   * Used to control the attributes of the visualization artifacts
+   */
+  double min_point_distance;      /* Minimum distance between consecutive trajectory points. */
+
+  //MoveIt! params
+  const double plan_time; //planning time of MoveIt!
+  const double step_size; 
+  const double jump_thresh;
+  const bool moveit_replan;
+  const bool avoid_collision;
+
+  const bool descartes_replan; //TOCHECK 
+  std::string descartes_planner_type; //dense vs sparse
+};
+
+
+/*  =============================== Hybrid Planner Class ===============================
+ *
+ * Provides a group of functions for planning and executing a robot path using Moveit and
+ * the Descartes Planning Library */
+
+class HybridPlanner
+{
+  public:
+
+    HybridPlanner();
+    virtual ~HybridPlanner();
+
+    void loadParameters();
+    void init();
+    void update();
+    // void initRos();
+    // void initDescartes();
+    void moveHome();
+    void generateTrajectory(descartes_traj& traj);
+    void planPath(descartes_traj& input_traj,descartes_traj& output_path);
+    void runPath();
+
+  protected:
+    // TOCHECK bool vs void. 
+    bool appendDescartesPoint();
+    bool appendDescartesCartPoint(Eigen::Affine3d& pose);
+    bool appendTolerancedDescartesCartPoint(Eigen::Affine3d& pose);
+    bool appendDescartesCartPointSegment(EigenSTL::vector_Affine3d poses);
+    bool appendDescartesJointPoint();
+    bool appendFreeSpaceSegment(std::vector<geometry_msgs::Pose> waypoints);
+    bool appendFreeSpaceCartPoint();
+    bool appendFreeSpaceJointPoint();
+    void planProcessPath();
+
+    // void HybridPlanner::appendDescartesCartPoint(EigenSTL::vector_Affine3d& pose);
+
+    /* Create an axial trajectory pt from a given tf transform */
+    descartes_core::TrajectoryPtPtr tfToAxialTrajectoryPt(const tf::Transform& nominal, double discretization);   
+
+    /*Translates a point relative to a reference pose to an absolute transformation
+    Also flips the z axis of the orientation to point INTO the plane specified by the
+    reference frame.*/
+    tf::Transform createNominalTransform(const geometry_msgs::Pose& ref_pose, const geometry_msgs::Point& point);
+
+    void HybridPlanner::fromDescartesToMoveitTrajectory();
+
+
+    void publishPosesMarkers(const EigenSTL::vector_Affine3d& poses);
```

Comment:
Okay. So I create a new class for execution and one for visualization. What else do you think is missing? Just asking to define the scope.